### PR TITLE
added support for authentication

### PR DESCRIPTION
**Pull request by mesudpasic**

### CODE REVIEW

File: Module.yaml [hunk 1]
```diff
@@ -28,6 +28,27 @@ environments:
...
+    value: ''
+    default: ''
+    type: string
+    name: Authentication token
+  - key: AUTHENTICATION_API_KEY
+    description: Authentication API key if required
+    value: ''
+    default: ''
+    type: string
+    name: Authentication API key
```
Comment (kubagrzelak) [inline Module.yaml]: We don't need this changes. Please, revert it.

File: README.md [hunk 2]
```diff
@@ -40,6 +40,9 @@ The following module configurations can be provided in a data service designer s
 | --------- | --------------------- | ------ | ------------------------------------------- |
 | Host Name | HOST_NAME             | string | Host name/address where the server will run |
 | Port      | HOST_PORT             | string | Port on which the server will run           |
+| AUTHENTICATION_REQUIRED          | enum | Determines if during request authentication is required           |
+| ACCESS_TOKEN          | string | If authentication is required this is the input for token, it also requires method like Basic or Bearer for example: Barer adhjaskjhjd-dsfdsfjsdkjf#42389dfsajfd==           |
+| AUTHENTICATION_API_KEY          | string | If we need to specify authentication X-API-TOKEN, here users can enter it|
```
Comment (kubagrzelak) [inline README.md]: We don't need this changes. Please, revert it.

File: config.env [hunk 3]
```diff
@@ -3,6 +3,8 @@
 MODULE_NAME=http-ingress
 MODULE_TYPE=INGRESS
 EGRESS_URL=https://hookb.in/YVDgXMdkmbsQERGGEQ99
-
+AUTHENTICATION_REQUIRED=no
+ACCESS_TOKEN=None
+AUTHENTICATION_API_KEY=None
```
Comment (kubagrzelak) [inline config.env]: We don't need this changes. Please, revert it.

File: image/src/app/config/application.py [hunk 4]
```diff
@@ -3,8 +3,10 @@
 """
 from app.utils.env import env
 
-
 APPLICATION = {
     "HOST_NAME": env("HOST_NAME", "0.0.0.0"),
     "HOST_PORT": env("HOST_PORT", "80"),
+    "AUTHENTICATION_REQUIRED": env("AUTHENTICATION_REQUIRED", "no"),
+    "ACCESS_TOKEN": env("ACCESS_TOKEN", ""),
+    "AUTHENTICATION_API_KEY": env("AUTHENTICATION_API_KEY", ""),
```
Comment (kubagrzelak) [inline image/src/app/config/application.py]: We don't need this changes. Please, revert it.

File: image/src/app/weeve/egress.py [hunk 5]
```diff
@@ -18,18 +19,27 @@ def send_data(data: json, timestamp=time.time()) -> bool:
     """
 
     try:
+        headers = {}
+        if APPLICATION['AUTHENTICATION_REQUIRED'] == 'yes' and APPLICATION['ACCESS_TOKEN'] != '':
+            headers.update({"Authorization": f"{APPLICATION['ACCESS_TOKEN']}"})
+        if APPLICATION['AUTHENTICATION_API_KEY'] != '':
+            headers.update({'x-api-key': APPLICATION['AUTHENTICATION_API_KEY']})
```
Comment (kubagrzelak) [inline image/src/app/weeve/egress.py]: We don't need this changes. Please, revert it.

File: image/src/app/weeve/egress.py [hunk 6]
```diff
@@ -18,18 +19,27 @@ def send_data(data: json, timestamp=time.time()) -> bool:
...
         # URL Convetion 1
         if not WEEVE['EGRESS_URL']:
-            resp = post(url=f"{WEEVE['EGRESS_SCHEME']}://{WEEVE['EGRESS_HOST']}:{WEEVE['EGRESS_PORT']}/{WEEVE['EGRESS_PATH']}", json=data)
+            resp = post(
+                url=f"{WEEVE['EGRESS_SCHEME']}://{WEEVE['EGRESS_HOST']}:{WEEVE['EGRESS_PORT']}/{WEEVE['EGRESS_PATH']}",
+                json=data,
+                headers=headers,
+            )
```
Comment (kubagrzelak) [inline image/src/app/weeve/egress.py]: We don't need this changes. Please, revert it.

File: image/src/app/weeve/egress.py [hunk 7]
```diff
@@ -18,18 +19,27 @@ def send_data(data: json, timestamp=time.time()) -> bool:
...
         # URL Convetion 2
         else:
-            resp = post(url=f"{WEEVE['EGRESS_URL']}", json=data)
-        
+            resp = post(url=f"{WEEVE['EGRESS_URL']}", json=data, headers=headers)
+
```
Comment (kubagrzelak) [inline image/src/app/weeve/egress.py]: We don't need this changes. Please, revert it.

**Review by kubagrzelak — CHANGES_REQUESTED** [review]
Hey, turns out there had to be some miscommunication and we do not need authentication for HTTP-Ingress module, because it POSTs data to another module in the edge application. This process does not require authentication, because it is happening within isolated Docker network. In this case, please revert all authetication changes and I can merge only the styling improvements. If you don't have time to be bothered with it, then we can just kill the PR

File: image/src/app/weeve/egress.py (hunk 7 above)
Comment (mesudpasic) [reply image/src/app/weeve/egress.py]: done.

File: image/src/app/weeve/egress.py (hunk 6 above)
Comment (mesudpasic) [reply image/src/app/weeve/egress.py]: done.

File: image/src/app/weeve/egress.py (hunk 5 above)
Comment (mesudpasic) [reply image/src/app/weeve/egress.py]: done.

File: image/src/app/config/application.py (hunk 4 above)
Comment (mesudpasic) [reply image/src/app/config/application.py]: done.

File: config.env (hunk 3 above)
Comment (mesudpasic) [reply config.env]: done.

File: README.md (hunk 2 above)
Comment (mesudpasic) [reply README.md]: done.

File: Module.yaml (hunk 1 above)
Comment (mesudpasic) [reply Module.yaml]: done.

**Review by kubagrzelak — APPROVED** [review]
Good job

**Pull request merged**